workbench-core/repo-scripts/github-oidc/src/github-oidc-stack.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CfnOutput, Stack, StackProps } from 'aws-cdk-lib';
+import { FederatedPrincipal, ManagedPolicy, OpenIdConnectProvider, Role } from 'aws-cdk-lib/aws-iam';
+import { NagSuppressions } from 'cdk-nag';
+import { Construct } from 'constructs';
+
+export interface GitHubOIDCStackProps extends StackProps {
+  gitHubOrg: string;
+  gitHubRepos: string[];
+  idp: OpenIdConnectProvider;
+}
+
+export class GitHubOIDCStack extends Stack {
+  public constructor(scope: Construct, id: string, props: GitHubOIDCStackProps) {
+    super(scope, id, props);
+
+    //Create GitHubOIDC Role
+    props.gitHubRepos.forEach((gitHubRepo) => {
+      const githubOIDCRole: Role = new Role(this, `${props.gitHubOrg}-${gitHubRepo}-GitHub-OIDC-Role`, {
```

Comment:
Is this name static or is it filled in by CDK during deploy? i.e. when deploying is it `${props.gitHubOrg}-${gitHubRepo}-GitHub-OIDC-Role` or does CDK add other characters as well

I ask to see if we should add names to a role at all

Reply:
This is an identifier for the `Role` and it is getting generated based on the values set in the [config.ts](https://github.com/aws-solutions/solution-spark-on-aws/blob/github-oidc/workbench-core/repo-scripts/github-oidc/src/configs/config.ts#L7) file. The role will be created for each repo in a GithubOrg. The name of the role generated will look like this: `aws-solutions-GitHubOIDCS-awssolutionssolutionspar-1U8F9SM4EZKLI`

Reply:
I am not specifying roleName explicitly to prevent tear down/update issues